src/mWater_collate/download_pictures.R

```diff
@@ -0,0 +1,107 @@
+library(tidyverse)
+
+download_pictures <- function(){
+  # Find all the downloaded pictures
+  all_file_names <- list.files(path = "data/field_pics/", recursive = TRUE)
+  #grab notes
+  sampling_photos <- all_notes%>%
+    #grab needed columns
+    select(site, start_dt,photos_downloaded,upstream_pic,downstream_pic,clarity,filter_pic,other_pic,other_pic_descriptor)%>%
+    mutate(
+      #Date format for pictures
+      yyyymmdd = format(start_dt, "%Y%m%d"),
+      #create filenames ONLY if there is a URL associated with the site visit
+      upstream_filename = case_when(
+        !is.na(upstream_pic) ~ paste0(site, "_", yyyymmdd, "_upstream.jpg"),
+        TRUE ~ NA_character_
+      ),
+      downstream_filename = case_when(
+        !is.na(downstream_pic) ~ paste0(site, "_", yyyymmdd, "_downstream.jpg"),
+        TRUE ~ NA_character_
+      ),
+      clarity_filename = case_when(
+        !is.na(clarity) ~ paste0(site, "_", yyyymmdd, "_clarity.jpg"),
+        TRUE ~ NA_character_
+      ),
+      filter_filename = case_when(
+        !is.na(filter_pic) ~ paste0(site, "_", yyyymmdd, "_filter.jpg"),
+        TRUE ~ NA_character_
+      ),
+      # check to see if the photo is already downloaded to folder
+      upstream_downloaded = case_when(
+        is.na(upstream_filename) ~ NA,
+        upstream_filename %in% all_file_names ~ TRUE,
+        TRUE ~ FALSE
+      ),
+      clarity_downloaded = case_when(
+        is.na(clarity_filename) ~ NA,
+        clarity_filename %in% all_file_names ~ TRUE,
+        TRUE ~ FALSE
+      ),
+      filter_downloaded = case_when(
+        is.na(filter_filename) ~ NA,
+        filter_filename %in% all_file_names ~ TRUE,
+        TRUE ~ FALSE
+      ),
+      downstream_downloaded = case_when(
+        is.na(downstream_filename) ~ NA,
+        downstream_filename %in% all_file_names ~ TRUE,
+        TRUE ~ FALSE
+      )
+    )
+# basic path to field pics
+path <- "data/field_pics/"
+
+
+  # loop thru dataset and download the photo ONLY if it is not yet downloaded and not NA
+  for (i in 1:length(sampling_photos$site)) {
```

Comment:
suggestion: If this is iterating over the number of rows in the df, I think it would be more clear to use `for (i in 1:nrow(sampling_photos))`.

Reply:
Will do, thanks!